pkg/cmd/cli/cmd/importimage.go

```diff
@@ -67,45 +68,147 @@ func RunImportImage(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, arg
 		return err
 	}
 
+	insecure := cmdutil.GetFlagBool(cmd, "insecure")
 	from := cmdutil.GetFlagString(cmd, "from")
 	confirm := cmdutil.GetFlagBool(cmd, "confirm")
-	insecure := cmdutil.GetFlagBool(cmd, "insecure-repository")
+	all := cmdutil.GetFlagBool(cmd, "all")
+
+	if len(from) == 0 {
+		from = streamName
+	}
+
+	ref, err := imageapi.ParseDockerImageReference(streamName)
+	if err != nil {
+		return fmt.Errorf("the image name must be a valid Docker image pull spec or reference to an image stream (e.g. myregistry/myteam/image:tag)")
+	}
+	if len(ref.ID) > 0 {
+		return fmt.Errorf("if you want to import an image by ID, use --from=%s %s", from, ref.AsRepository().Exact())
+	}
+	// apply the default tag
+	if !all && len(ref.Tag) == 0 {
+		ref.Tag = imageapi.DefaultImageTag
+	}
+	name := ref.Name
+
 	imageStreamClient := osClient.ImageStreams(namespace)
-	stream, err := imageStreamClient.Get(streamName)
+	stream, err := imageStreamClient.Get(name)
```

Comment:
This whole section could use refactoring.  This chunk is about getting an `ImageStream` for a given name that can be either the actual resource or a theoretical one?